src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -391,9 +391,10 @@ def _merge_input_ids_with_image_features(
         # `special_image_token_mask` identifies image tokens. Each image token will be replaced by `nb_text_tokens_per_images - 1` text tokens.
         # `torch.cumsum` computes how each image token shifts subsequent text token positions.
         # - 1 to adjust for zero-based indexing, as `cumsum` inherently increases indices by one.
-        patches_lengths = torch.tensor(patches_lengths, device=inputs_embeds.device).unsqueeze(dim=1)
-        repeated_patches = patches_lengths.repeat(1, special_image_token_mask.shape[1]) 
-        new_token_positions = torch.cumsum(special_image_token_mask * (repeated_patches - 1) + 1, -1) - 1
+        patches_lengths = torch.tensor(patches_lengths, device=inputs_embeds.device)
```

Comment:
In patches_lengths, how are we distinguishing between the lengths of the various images in each input?
Am I assuming correctly that right now it is the sum of image features across all  input images per input?

Reply:
all of them were padded to the same size after tokenization but we unpad them in the forward() call

Reply:
We don't distinguish between sequences, since it's not necessary to keep that info for filling in the image embeddings later.

Reply:
ah makes sense. Nice, this is way more straightforward than the original code!